openmc/material.py

```diff
@@ -908,22 +899,57 @@ def get_nuclide_atom_densities(self):
 
         return nuclides
 
-    def get_nuclide_activity(self):
-        """Return activity in [Bq] for each nuclide in the material
+    def get_activity(self, normalization: str = 'total', by_nuclide: bool = False):
+        """Returns the activity of the material or for each nuclide in the
+        material in units of [Bq], [Bq/g] or [Bq/cc].
 
         .. versionadded:: 0.13.1
 
+        Parameters
+        ----------
+        normalization : {'total', 'mass', 'volume'}
+            Specifies the type of activity to return, 'total' will return the
+            material activity in [Bq], 'mass' returns the materials specific
+            activity in [Bq/g] and 'volume' returns the activity in [Bq/cc].
+        by_nuclide : bool
+            Specifies if the activity should be returned for the material as a
+            whole or per nuclide.
+
         Returns
         -------
-        dict
-            Dictionary whose keys are nuclide names and values are activity in
-            [Bq].
+
+        Union[dict, float]
+            If by_nuclide is True then a dictionary whose keys are nuclide 
+            names and values are activity is returned. Otherwise the activity
+            of the material is returned as a float.
         """
-        activity = {}
-        for nuclide, atoms in self.get_nuclide_atoms().items():
-            inv_seconds = openmc.data.decay_constant(nuclide)
-            activity[nuclide] = inv_seconds * atoms
-        return activity
+
+        cv.check_value('normalization', normalization, {'total', 'mass', 'volume'})
+        cv.check_type('by_nuclide', by_nuclide, bool)
+        
+        if normalization=='total':
+            activity = {}
+            for nuclide, atoms in self.get_nuclide_atoms().items():
+                inv_seconds = openmc.data.decay_constant(nuclide)
+                activity[nuclide] = inv_seconds * atoms
+
+        elif normalization=='mass':
+            activity = {}
+            for nuclide, atoms in self.get_nuclide_atom_densities().items():
+                inv_seconds = openmc.data.decay_constant(nuclide)
+                activity[nuclide] = (inv_seconds * atoms * 1.0e24) / self.density
+        # normalization must be volume by this stage so else can be used
+        else:
+            activity = {}
+            for nuclide, atoms in self.get_nuclide_atom_densities().items():
+                inv_seconds = openmc.data.decay_constant(nuclide)
+                activity[nuclide] = (inv_seconds * atoms * 1.0e24) / self.volume
```

Comment:
```suggestion
        if normalization == 'total':
            multiplier = self.volume
        else:
            multiplier = 1 if normalization == 'volume' else 1.0 / self.get_mass_density()
            
        activity = {}
        for nuclide, atoms_per_bcm in self.get_nuclide_atom_densities().items():
            inv_seconds = openmc.data.decay_constant(nuclide)
            activity[nuclide] = inv_seconds * 1e24 * atoms_per_bcm * multiplier
```

Reply:
I think a few of those calculations aren't quite correct and the code could be condensed a bit.

Reply:
I think normalizing by volume as default makes the most sense since you don't need to set the volume property.